doc/report/TUDreport.tex

```diff
@@ -0,0 +1,538 @@
+    \footnote{\url{https://github.com/strongloop/loopback-example-passport}}. It setups some preliminary routes for \emph{Express.js} as well as configuring some middleware components, e.g., to parse form data. Eventually, the start-up routine is invoked and forwards all requests to \emph{LoopBack} which than handles the request accordingly.
+
+    The purpose of \emph{StrongLoop} is to provide a REST API to do basic CRUD\footnote{Create/Read/Update/Delete operations on a persistent storage} operations on a predefined database schema. Since all the magic is done by \emph{StrongLoop}, a developer usually does not need to dive further into the topic of request handling etc. Everything that a developer needs to do is defining and/or extending the models inside \emph{/common/models} and \emph{StrongLoop} will take care of the rest. The process of model creation and modification is described later in section~\ref{sec:tech:implementation:strongloop:models}.
+
+  \subsection{Web Application}
+  \label{sec:tech:implementation:bootstrapping:web}
+
+    As previously mentioned~(\ref{sec:tech:implementation:dirs}), the \emph{/client} directory is exposed through the HTTP web server. In particular, the client is eventually sending an HTTP request to the root of this directory. Because no particular file is requested by the client, the server will respond with the contents of the \emph{index.html} file.
+
+    The \emph{index.html} file consists of a simple HTML5 skeleton which basically includes lots of CSS and JavaScript files. One of the libraries that gets included by the file is \emph{AngularJS}. Once the browser is done with parsing the page and loading all embedded JavaScript files, the \emph{AngularJS} framework is scanning for an \texttt{ng-app} attribute inside the DOM. When found, AngularJS automatically starts its bootstrap routine and eventually executes the application, which is happen to be defined inside \emph{js/app.js}. Section~\ref{sec:tech:implementation:angularjs} covers the contents of this file.
+
+  \section{StrongLoop}
+  \label{sec:tech:implementation:strongloop}
+
+    As explained in section~\ref{sec:tech:architecture:3rd-party}, \emph{StrongLoop} is the tool that the application uses to execute CRUD operations on the database. Besides utilizing \emph{StrongLoop} as an ORM, the application makes also use of its other features such as authentication- \& session management, access control and file upload support. The application is mainly managed by the command line tool, \texttt{slc}, that comes together with \emph{StrongLoop}. It allows to generate and extend models, define access control rules and more things. For an exhaustive description of the tool please refer to the official documentation.\footnote{\url{https://docs.strongloop.com/display/public/LB/Command-line+reference}}
```

Comment:
- statt more things würde ich much more schreiben
- .. of the tool, please refer.. (komma)